### PR TITLE
feature/214 - fix reporting failed workflow due to failed step

### PR DESCRIPTION
**Pull request by lucamrgs**
To fix this, I changed the logic of assignment of the ID of the next step to execute.
The new logic is more in line with a consistent use of the CACAO V2 spec.

### CODE REVIEW

**Comment by lucamrgs** [comment]
The updated "next step ID assignment" based on on_completion (and ignoring on_success and on_failure) implies that we should not use on_success and on_failure at all at the moment - which I think is a necessary decision for CACAO V2. To be changed in CACAO V3.

**Comment by MaartendeKruijf** [comment]
This would prepare for the new CACAO standard right? The question I've wouldn't this break current playbooks that use OnSucces?

**Comment by lucamrgs** [comment]
@MaartendeKruijf it's not a change meant to align with the next CACAO version, but to a change that actually records when a playbook execution fails. A failed playbook execution happens when an execution error arises, not when a step "fails". This was overlooked in the code that this PR changes. It would not actually really break playbooks that used on_success, because as far as our execution goes (and if I am not mistaken), we would only "branch" into on_faliure IF on_faliure was the only one defined among on_completion, on_success, and on_faliure. 

It is true though that before our logic was defaulting on on_success being the only one defined - but now it's just switched around on defaulting on on_completion.

In summary, this PR should just fix previois incorrect behaviour/assumptions.

**Review by MaartendeKruijf — APPROVED** [review]
